pkg/config/validation/validation.go

```diff
@@ -1129,33 +1129,45 @@ func validateLoadBalancer(settings *networking.LoadBalancerSettings) (errs error
 	return
 }
 
-func validateTLS(settings *networking.ClientTLSSettings) (errs error) {
+func validateTLS(settings *networking.ClientTLSSettings) Validation {
+	v := Validation{}
 	if settings == nil {
-		return
+		return v
 	}
 
 	if (settings.Mode == networking.ClientTLSSettings_SIMPLE || settings.Mode == networking.ClientTLSSettings_MUTUAL) &&
 		settings.CredentialName != "" {
 		if settings.ClientCertificate != "" || settings.CaCertificates != "" || settings.PrivateKey != "" {
-			errs = appendErrors(errs,
+			v = appendValidation(v,
 				fmt.Errorf("cannot specify client certificates or CA certificate If credentialName is set"))
 		}
 
 		// If tls mode is SIMPLE or MUTUAL, and CredentialName is specified, credentials are fetched
 		// remotely. ServerCertificate and CaCertificates fields are not required.
-		return
+		return v
 	}
 
 	if settings.Mode == networking.ClientTLSSettings_MUTUAL {
 		if settings.ClientCertificate == "" {
-			errs = appendErrors(errs, fmt.Errorf("client certificate required for mutual tls"))
+			v = appendValidation(v, fmt.Errorf("client certificate required for mutual tls"))
 		}
 		if settings.PrivateKey == "" {
-			errs = appendErrors(errs, fmt.Errorf("private key required for mutual tls"))
+			v = appendValidation(v, fmt.Errorf("private key required for mutual tls"))
+		}
+		if settings.CaCertificates == "" {
+			v = appendValidation(v, WrapWarning(fmt.Errorf("ca certificate required for mutual tls,  it will fail in release 1.12")))
```

Comment:
Is it actually required? or does not setting it allow us to have optional verification? ie equivalent to golang `tls.RequestClientCert`. https://github.com/istio/istio/issues/31984

Reply:
make sense

Reply:
Nope, this is mutual tls, not ClientTLSSettings_SIMPLE

Reply:
Yes I know. You can have mutual TLS without caCert (in theory - I don't know if it works in Istio today). See the linked issue

Reply:
basically optional mtls